lib/src/value/color.ts

```diff
@@ -208,166 +348,113 @@ export class SassColor extends Value {
+        ];
+        otherCoords = other.color.to('srgb').coords.map(coordToRgb) as [
+          number,
+          number,
+          number,
+        ];
+      }
+      return (
+        fuzzyEquals(coords[0], otherCoords[0]) &&
+        fuzzyEquals(coords[1], otherCoords[1]) &&
+        fuzzyEquals(coords[2], otherCoords[2])
+      );
+    }
     return (
-      other instanceof SassColor &&
-      fuzzyEquals(this.red, other.red) &&
-      fuzzyEquals(this.green, other.green) &&
-      fuzzyEquals(this.blue, other.blue) &&
+      this.space === other.space &&
+      fuzzyEquals(coords[0], otherCoords[0]) &&
+      fuzzyEquals(coords[1], otherCoords[1]) &&
+      fuzzyEquals(coords[2], otherCoords[2]) &&
       fuzzyEquals(this.alpha, other.alpha)
     );
   }
 
   hashCode(): number {
-    return hash(this.red ^ this.green ^ this.blue ^ this.alpha);
-  }
-
-  toString(): string {
-    const isOpaque = fuzzyEquals(this.alpha, 1);
-    let string = isOpaque ? 'rgb(' : 'rgba(';
-    string += `${this.red}, ${this.green}, ${this.blue}`;
-    string += isOpaque ? ')' : `, ${this.alpha})`;
-    return string;
-  }
-
-  // Computes `this`'s `hue`, `saturation`, and `lightness` values based on
-  // `red`, `green`, and `blue`.
-  //
-  // Algorithm from https://en.wikipedia.org/wiki/HSL_and_HSV#RGB_to_HSL_and_HSV
-  private rgbToHsl(): void {
-    const scaledRed = this.red / 255;
-    const scaledGreen = this.green / 255;
-    const scaledBlue = this.blue / 255;
-
-    const max = Math.max(scaledRed, scaledGreen, scaledBlue);
-    const min = Math.min(scaledRed, scaledGreen, scaledBlue);
-    const delta = max - min;
-
-    if (max === min) {
-      this.hueInternal = 0;
-    } else if (max === scaledRed) {
-      this.hueInternal = positiveMod(
-        (60 * (scaledGreen - scaledBlue)) / delta,
-        360
-      );
-    } else if (max === scaledGreen) {
-      this.hueInternal = positiveMod(
-        120 + (60 * (scaledBlue - scaledRed)) / delta,
-        360
+    let coords = this.color.coords;
+    if (this.isLegacy) {
+      coords = this.color.to('srgb').coords.map(coordToRgb) as [
+        number,
+        number,
+        number,
+      ];
+      return (
+        fuzzyHashCode(coords[0]) ^
+        fuzzyHashCode(coords[1]) ^
+        fuzzyHashCode(coords[2]) ^
+        fuzzyHashCode(this.alpha)
       );
-    } else if (max === scaledBlue) {
-      this.hueInternal = positiveMod(
-        240 + (60 * (scaledRed - scaledGreen)) / delta,
-        360
-      );
-    }
-
-    this.lightnessInternal = 50 * (max + min);
-
-    if (max === min) {
-      this.saturationInternal = 0;
-    } else if (this.lightnessInternal < 50) {
-      this.saturationInternal = (100 * delta) / (max + min);
-    } else {
-      this.saturationInternal = (100 * delta) / (2 - max - min);
     }
+    return (
+      fuzzyHashCode(coords[0]) ^
+      fuzzyHashCode(coords[1]) ^
+      fuzzyHashCode(coords[2]) ^
+      fuzzyHashCode(this.alpha)
+    );
```

Comment:
I see that the Dart Sass implementation also includes `space.hashCode` here -- I assume that's something we need here too? If so, how is it calculated?

Reply:
In Dart, `hashCode` is defined for every object and just defaults to being based on identity (like `operator==`). I believe ImmuableJS's [`hash()` function](https://immutable-js.com/docs/latest@main/hash()/) will hash strings based on contents.

Reply:
Addressed in https://github.com/oddbird/embedded-host-node/commit/133dd7b52e26489de72f55e7032c4cad211077e6